airbyte/sources/base.py

```diff
@@ -75,6 +75,7 @@ def __init__(
         self.executor = executor
         self.name = name
         self._processed_records = 0
+        self._last_stream_processed: str = None
```

Comment:
Especially with parent-child streams, some sources may bounce between different stream names. Wdyt of tracking this as a set?

```suggestion
        self._streams_processed: set[str] = set()
```